src/api/routes/transcripts.ts

```diff
@@ -52,8 +53,65 @@ const list = procedure
   return g.transcripts;
 });
 
+const getNameMap = procedure
+  .use(authUser())
+  .input(z.string())
+  .query(async ({ input: transcriptId }) => {
+
+    const summaries = await db.Summary.findAll({ where: { transcriptId } });
+
+    // find all handlebar from summaries under one transcript
+    let handlebars: string[] = [];
+    for (let s of summaries) {
+      const matches = s.summary.match(/{{(.*?)}}/g);
+      if (matches) {
+        handlebars = [...handlebars, ...matches.map(match => match.slice(2, -2).trim())];
+      }
+    }
+
+    let nameMap: Record<string, string> = {};
+    // create a list of namemap of itself, otherwise when handlebars compile it will return empty
+    nameMap = handlebars.reduce((o, key) => ({ ...o, [key]: key }), {});
+
+    const tnm = await db.TranscriptNameMapping.findAll({
+      where: { handlebarName: handlebars },
+      include: [{
+        model: db.User,
+        attributes: ['name']
+      }],
+    });
+
+    if (tnm) {
+      for (const nm of tnm) {
+        invariant(nm.user.name);
+        nameMap[nm.handlebarName] = nm.user.name;
+      }
+    }
+
+    return nameMap;
+
+  });
+
+const updateNameMap = procedure
```

Comment:
it doesn't belong here. create a new file (and router entry) `transcriptNameMapping.ts`.

also, use full names and avoid short names for symbols with large scope.